gems/pending/middleware/events/middleware_event_monitor.rb

```diff
@@ -0,0 +1,31 @@
+class MiddlewareEventMonitor
+  def initialize(ems, start_time)
+    @ems = ems
+    @alerts_client = ems.alerts_connect
+    # TODO: can we store and resume where we left off?  To avoid duplicates, only look for events now or later,
+    # so basically we miss events when we're not running...
```

Comment:
Moving forward I will likely try and improve the whole mechanism for catching the events, this simple poll has various issues, like the one mentioned.